src/app/ReadClient.cpp

```diff
@@ -412,13 +406,16 @@ CHIP_ERROR ReadClient::ProcessReportData(System::PacketBufferHandle && aPayload)
         }
     }
 
-    StatusResponse::SendStatusResponse(err == CHIP_NO_ERROR ? Protocols::InteractionModel::Status::Success
-                                                            : Protocols::InteractionModel::Status::InvalidSubscription,
-                                       mpExchangeCtx, IsAwaitingSubscribeResponse() || mPendingMoreChunks);
-
-    if (!mInitialReport && !mPendingMoreChunks)
+    if (!suppressResponse)
     {
-        mpExchangeCtx = nullptr;
+        err = StatusResponse::SendStatusResponse(err == CHIP_NO_ERROR ? Protocols::InteractionModel::Status::Success
+                                                                      : Protocols::InteractionModel::Status::InvalidSubscription,
+                                                 mpExchangeCtx, IsAwaitingSubscribeResponse() || mPendingMoreChunks);
+
+        if ((!mInitialReport && !mPendingMoreChunks) || (err != CHIP_NO_ERROR))
```

Comment:
Why is the check here different from the last arg passed to SendStatusResponse?  `!mInitialReport` here vs `IsAwaitingSubscribeResponse()` there.

Reply:
just fixed this issue and introduce bool notExpectResponse = IsSubscriptionIdle() && !mPendingMoreChunks; which mean not expect response for status response delivery when processing last non-chunk non-solicit report